File: samples/java/confluent/src/main/java/com/contoso/kafka/KafkaOutputMany.java
// package com.contoso.kafka;

// import java.util.*;
// import com.microsoft.azure.functions.annotation.*;
// import com.microsoft.azure.functions.*;

// import java.util.Optional;

// public class KafkaOutputMany {
//     @FunctionName("KafkaOutputMany")
//     public HttpResponseMessage run(
//             @HttpTrigger(name = "req", methods = {HttpMethod.GET, HttpMethod.POST}, authLevel = AuthorizationLevel.ANONYMOUS) HttpRequestMessage<Optional<String>> request,
//             @KafkaOutput(
//                 name = "kafkaOutput",
//                 topic = "message", 
//                 brokerList="%BrokerList%",
//                 username = "%ConfluentCloudUsername%", 
//                 password = "%ConfluentCloudPassword%",
//                 authenticationMode = BrokerAuthenticationMode.PLAIN,
//                 // sslCaLocation = "confluent_cloud_cacert.pem", // Enable this line for windows.  
//                 protocol = BrokerProtocol.SASLSSL
//             )  OutputBinding<String[]> output,
//             final ExecutionContext context) {
//         context.getLogger().info("Java HTTP trigger processed a request.");
//         String[] messages = new String[2];
//         messages[0] = "one";
//         messages[1] = "two";
//         output.setValue(messages);
//         return request.createResponseBuilder(HttpStatus.OK).body("Ok").build();
//     }
// }

File: samples/dotnet/EventHub/KafkaOutputMany.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.Azure.WebJobs.Extensions.Kafka;
using Microsoft.Extensions.Logging;

namespace Eventhub
{
    public class KafkaOutputMany
    {
        [FunctionName("KafkaOutputMany")]
        public static IActionResult Output(
            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequest req,
            [Kafka("BrokerList",
                    "topic",
                    Username = "$ConnectionString",
                    Password = "%EventHubConnectionString%",
                    Protocol = BrokerProtocol.SaslSsl,
                   AuthenticationMode = BrokerAuthenticationMode.Plain
            )] out KafkaEventData<string>[] eventDataArr,
            ILogger log)
        {
            log.LogInformation("C# HTTP trigger function processed a request.");
            eventDataArr = new KafkaEventData<string>[2];
            eventDataArr[0] = new KafkaEventData<string>("one");
            eventDataArr[1] = new KafkaEventData<string>("two");
            return new OkObjectResult("Ok");
        }
    }
}